service-registry/src/main/java/org/apache/servicecomb/serviceregistry/registry/AbstractServiceRegistry.java

```diff
@@ -303,4 +307,30 @@ public void destroy() {
     eventBus.post(new ShutdownEvent());
     unregisterInstance();
   }
+
+  @Override
+  public void registryMicroserviceMapping(String microserviceName, String version, Class<?> schemaIntfCls,
+      List<MicroserviceInstance> instances) {
+    String app = RegistryUtils.getAppId();
+    MicroserviceManager microserviceManager = appManager.getOrCreateMicroserviceManager(app);
+    StaticMicroserviceVersions microserviceVersions =
+        (StaticMicroserviceVersions) microserviceManager.getVersionsByName()
+            .computeIfAbsent(microserviceName,
+                svcName -> new StaticMicroserviceVersions(this.appManager, app, microserviceName, schemaIntfCls));
+
+    microserviceVersions.addInstances(version, instances);
```

Comment:
addInstances should put inside computeIfAbsent?

Reply:
`addInstances` is put inside `computeIfAbsent`. Currently only initial instance method is provided. Users cannot re-set instance info of a 3rd party service.